Source/diablo.cpp

```diff
@@ -48,6 +51,21 @@ int gnTickRate;
 WORD gnTickDelay = 50;
 /** Game options */
 Options sgOptions;
+Keymapper keymapper {
+	// Workaround: remove once the INI library has been replaced.
+	[](const std::string &key, const std::string &value)
+	{
+		setIniValue("Controller.Keymapping", key.c_str(), value.c_str());
+	},
+	[](const std::string &key) -> std::string

```

Comment:
This PR introduces a lot of new concepts to the code base, keep in mind that DevilutionX is how I learned C++ and since it's mostly just C I don't know a lot of C++. I have no idea what I'm looking at here, I'm guessing this is an arrow function,  but what does the square brackets at the start of the line do?

Reply:
I think it's https://en.cppreference.com/w/cpp/language/lambda (I'm in the same boat as you : P)

Reply:
Yes its a lambda, an anonymous function: https://stackoverflow.com/questions/7627098/what-is-a-lambda-expression-in-c11 One of the best additions brought by C++11 :slightly_smiling_face:

Reply:
I like anonymous functions, so not saying we shouldn't use them, but just expect that it will take a bit longer than usual before this gets merged as I have to read up on things a bit. Sorry for gatekeeping, and any rebases that you might have to ensure :)

Reply:
No worries. It makes sense that you want to understand code before you merge it. :+1: Feel free to ask any question (well, C++ related at least :P).